ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -41,15 +38,13 @@
    * @throws StoreException when {@link BlobStore} is not started
    */
   CompactionDetails getCompactionDetails(BlobStore blobStore) throws StoreException {
-    long usedCapacity = blobStore.getSizeInBytes();
-    long totalCapacity = blobStore.getCapacityInBytes();
-    CompactionDetails details = null;
-    if (usedCapacity >= (storeConfig.storeMinUsedCapacityToTriggerCompactionInPercentage / 100.0) * totalCapacity) {
-      List<String> potentialLogSegments = blobStore.getLogSegmentsNotInJournal();
-      if (potentialLogSegments != null) {
-        details = new CompactionDetails(time.milliseconds() - messageRetentionTimeInMs, potentialLogSegments);
-      }
+    CompactionPolicyFactory compactionPolicyFactory;
```

Comment:
The instantiation of `CompactionPolicyFactory` could happen in the constructor, right? Since the factory type won't change while running